utils/mil_tools/nodes/label_me_bag.py

```diff
@@ -0,0 +1,268 @@
+#!/usr/bin/env python
+
+"""
+Converts bags to labelme images, or labelme annotations to bags.
+
+Uses a YAML file to specify what times and topics in a set of bag files
+will be used to produce images which are places in a corosponding
+folder in a LabelMe instalation.
+
+When run with the -e flag, uses same YAML file to read LabelMe annotations
+and place them as a ros message at syncronized times with the images
+they corospond to.
+
+With the -r flag, a report will be generated based on the specified YAML
+about how many images are labeled in LabelMe
+
+"""
+
+import argparse
+import yaml
+import xml.etree.ElementTree
+import os
+import rosbag
+import rospy
+import cv2
+import genpy
+from cv_bridge import CvBridge
+from geometry_msgs.msg import Point
+from mil_msgs.msg import LabeledObjects, LabeledObject
+
+class BagToLabelMe():
+  """
+  Interfaces between ROS bags and LabelMe images.
+  
+  Can be used to insert images from ROS bags into LabelMe instalation
+  or extract LabelMe annotations into a ROS bag
+  
+  TODO: improve verbosity, disable print statements with toggle
+  """
+  def __init__(self, config, labelme_dir, verbose=False):
+      """
+      Generates class that can be used to insert or extract images to/from LabelMe
+      
+      config: path to YAML file following format of the example yaml file in this package
+      labelme_dir: root directory of labelme instalation (should contain Images and Annotations directories)
+      """
+      self.verbose = verbose
+      if not os.path.isdir(labelme_dir):
+          raise Exception("Labelme directory {} does not exsist".format(labelme_dir))
+
+      config_file = open(config, 'r')
+      self.config = yaml.load(config_file)
+      self._verify_yaml()
```

Comment:
```
Use one leading underscore only for non-public methods and instance variables.

To avoid name clashes with subclasses, use two leading underscores to invoke Python's name mangling rules.
```